packages/web-components/src/rating-display/rating-display.template.ts

```diff
@@ -0,0 +1,51 @@
+import { ElementViewTemplate, html, repeat, when } from '@microsoft/fast-element';
+import type { RatingDisplay } from './rating-display.js';
+
+/**
+ * Reusable star icon symbol
+ */
+const star = html`
+  <svg xmlns="http://www.w3.org/2000/svg" style="display: none">
+    <symbol id="star" viewBox="0 0 12 12">
+      <path
+        d="M5.28347 1.54556C5.57692 0.95096 6.42479 0.950961 6.71825 1.54556L7.82997 3.79817L10.3159 4.15939C10.9721 4.25474 11.2341 5.06112 10.7592 5.52394L8.96043 7.27736L9.38507 9.75321C9.49716 10.4067 8.81122 10.9051 8.22431 10.5966L6.00086 9.42761L3.7774 10.5966C3.19049 10.9051 2.50455 10.4067 2.61664 9.75321L3.04128 7.27736L1.24246 5.52394C0.767651 5.06111 1.02966 4.25474 1.68584 4.15939L4.17174 3.79817L5.28347 1.54556Z"
+      />
+    </symbol>
+  </svg>
+`;
+
+/**
+ * Generates a template for the Rating Display component.
+ *
+ * @public
+ */
+export function ratingDisplayTemplate<T extends RatingDisplay>(): ElementViewTemplate<T> {
+  return html<T>`
+    <template role="img" aria-labelledby=${x => `${x.uid}-value${x.count ? ` ${x.uid}-count` : ''}`}>
+      ${star}
+      ${repeat(
```

Comment:
Rating won't be cheap with `repeat()` here, but off the top of my head I'm not sure a great way to handle this given we need to observe selected.

Reply:
We also need to generate the elements dynamically here because the number of icons is determined by the `max` attribute.

Reply:
The only other option I can think of here would be a pretty big departure from the current strategy. Instead of repeating a `max` # of `<svg><use></use></svg>` calls, we could maybe draw a single SVG and use [`<pattern>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/pattern) to tile the star like a background image? 

You might have to composite filled stars and unfilled stars on top of each other to achieve the half-star effect...

```svg
<svg aria-hidden="true">
  <pattern id="star">...</pattern>
  <pattern id="starFill">...</pattern>
  <rect x="0" y="0" fill="url(#star)" width="${ count * max }" height="${size}">
  <rect x="0" y="0" fill="url(#starFill)" width="${ size * icons }" height="${size}">
</svg>
```

Here's a quick prototype of that:

https://codepen.io/davatron5000/pen/MWdQEaJ

You'd also need to account for the `gap` which gets a bit harder 🫠 It's possible you'd do enough interpolation here that it'd perform worse than a `repeat`. Maybe we save this for a later investigation or enhancement. As it is currently the repeated icons aligns with the pattern in React.

Reply:
I wonder if it makes sense to perf test this just to be sure. Here's the code for repeat (without linking out to every dependency of it): https://github.com/microsoft/fast/blob/master/packages/web-components/fast-element/src/templating/repeat.ts

Reply:
Thanks for the suggestion, @davatron5000. I haven't tried this exact solution, drawing a single svg seems to be more performant compared to some other approaches I've tried. But like you said, it gets quite complicated once the gap is added, especially that the gap and the icon size will change based on the `size` attribute. I'm happy to try out this approach as an enhancement.

I'm currently trying out something that doesn't deviate much from this PR but without using `repeat`. Let's see how it goes.

Thanks for sharing the link @chrisdholt. It would be great if we can perf test this component. Is there a way to do it?

Reply:
I've pushed a commit removing the dependency on `repeat` and using our own code to create the icons based on the `max` attribute. The new method only renders the elements and doesn't have the other functionalities of the `repeat` helper that are not needed by this component. We ended up with less code in the class too.